types/decimal_test.go

```diff
@@ -243,6 +245,21 @@ func (s *decimalTestSuite) TestArithmetic() {
 	}
 }
 
+func (s *decimalTestSuite) TestMulRoundUp_RoundingAtPrecisionEnd() {
+	var (
+		a                = sdk.MustNewDecFromStr("0.000000000000000009")
+		b                = sdk.MustNewDecFromStr("0.000000000000000009")
+		expectedRoundUp  = sdk.MustNewDecFromStr("0.000000000000000001")
+		expectedTruncate = sdk.MustNewDecFromStr("0.000000000000000000")
```

Comment:
I think I might be misunderstanding this. a*b is 1.8*10-17, and rounding this up would be 2 right?

Reply:
a * b = 8.1 * 10^-35
```python
>>> a = 9 * Decimal("10")**(-18)
>>> a * a
Decimal('8.1E-35')
```

So we chop precision and round up at the end leading to the minimum non-zero value of `sdk.Dec` of 10^-18.

On the contrary, with truncation, we just chop precision and round down leading to zero 

Let me know if that makes sense

Reply:
Lol yeah, I did the math wrong, thanks